### PR TITLE
[#498] Implement message format

### LINKED ISSUE #498: Define a message format
A definition of the message format should be defined. A google docs should be created (This one can be updated : https://docs.google.com/spreadsheets/u/1/d/1bE3h0ZCqPAUhW6Gn6G0fKEoOPdopGTZnmmWK1VuVurI/edit?usp=drive_web&ouid=111054195812684486479)

The message format should tell us exactly what information is contained, the sheet should make sure that we have the information of the command ID of each command, their description, arguments, argument type etc. So that other members can easily refer to that sheet.

### PR DESCRIPTION
**Pull request by VimFreeman**
# Assignee Section

## Description
Added message format

### Steps for Testing
Check if GitHub CI runs

closes #498 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [ ] Local Test Performed Successfully
- [ ] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

**Review by MartensCedric — DISMISSED** [review]

**Review by ginywiny — COMMENTED** [review]
Everything looks good to be pushed for further development. Upon building, I didn't have any warnings either.

**Review by ginywiny — APPROVED** [review]

File: robot/rover/comms/src/CommandCenter.cpp [hunk 1]
```diff
@@ -9,21 +9,38 @@ namespace internal_comms
     Command* CommandCenter::processCommand() const
     {
         uint8_t commandID = Serial.read();
-        size_t bytesToRead = Serial.read();
-        uint8_t* buffer = (uint8_t*) malloc(sizeof(uint8_t) * bytesToRead);
-        uint8_t bytesRead = (uint8_t) Serial.readBytes((char*)buffer, bytesRead);
+        uint8_t deviceSending = Serial.read();
+        uint8_t deviceReceiving = Serial.read();
+        uint16_t messageLength = readMessageSize();
```
Comment (ginywiny) [inline robot/rover/comms/src/CommandCenter.cpp:14]: This is the length of the entire message though as opposed to just the argument length, is it not? If it is the argument length, can this be renamed to something more suitable like argumentLength?

Comment (VimFreeman) [reply robot/rover/comms/src/CommandCenter.cpp:14]: Its the message within the message 👀

Comment (VimFreeman) [reply robot/rover/comms/src/CommandCenter.cpp:14]: Its... Inmessaption

Comment (ginywiny) [reply robot/rover/comms/src/CommandCenter.cpp:14]: Yeah, I had to reread the readMessageSize() implementation. Would be good to rename to argumentLength.

Comment (VimFreeman) [reply robot/rover/comms/src/CommandCenter.cpp]: I changed the names

**Review by MartensCedric — APPROVED** [review]

File: robot/rover/comms/src/CommandCenter.cpp [hunk 2]
```diff
@@ -9,21 +9,39 @@ namespace internal_comms
...
-        com->isValid = true;
-        com->rawArgs = buffer;
-        com->rawArgsLength = bytesRead;
-        return com;
+    uint16_t CommandCenter::readArgSize() const 
+    {
+        // Serial.read() returns size_t
+        uint16_t byte1 = Serial.read();
+        uint8_t byte2 = Serial.read();
+        uint16_t ArgumentsLength = (byte1 << 8) | byte2;
```
Comment (VimFreeman) [inline robot/rover/comms/src/CommandCenter.cpp:44]: Prob gonna fail cause `read()` returns 16bits

**Review by VimFreeman — COMMENTED** [review]
1. We should have a global baudrate set in CommandCenter.h
2. We can go way higher than 9600

**Comment by ginywiny** [comment]
Add Ping message to all module types (Wheels, Arm, Science, PDS). Should it be generalized though? Not sure

**Pull request merged**